sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -792,3 +792,82 @@ public bool TryAdd(TKey key, TValue value)
         }
     }
 }
+
+namespace CSharp8
+{
+    public class NullCoalescenceAssignment
+    {
+        public void Test()
+        {
+            List<int> list = null;
+            list ??= new List<int>();
+            list.Clear(); // Noncompliant
+        }
+    }
+
+    public class SwitchExpression
+    {
+        private static bool Predicate(int i) => true;
+
+        public bool InsideSwitch(int type)
+        {
+            var list = new List<int>();
+
+            return type switch
+            {
+                1 => list.Exists(Predicate), // Noncompliant
+                _ => false
+            };
+        }
+
+        public void UsingSwitchResult(bool cond)
+        {
+            var list = cond switch
+            {
+                _ => new List<int>()
+            };
+
+            list.Clear(); // Noncompliant
+        }
```

Comment:
I think it would be nice to add a compliant example when there are multiple arms:
```
        public void UsingSwitchResult_Compliant(bool cond)
        {
            IList<int> list = cond switch
            {
                true => new int[] { 5 },
                _ => new List<int>()
            };

            list.Clear();
        }
```